src/Dfe.PlanTech.Web/ViewComponents/CategorySectionViewComponent.cs

```diff
@@ -95,14 +121,68 @@ public async Task<Category> RetrieveSectionStatuses(Category category)
     {
         try
         {
-            category.SectionStatuses = await _query.GetSectionSubmissionStatuses(category.Sections);
+            category.SectionStatuses = await _query.GetSectionSubmissionStatuses(category.Sys.Id);
+            category.Completed = category.SectionStatuses.Count(x => x.Completed == 1);
+            category.RetrievalError = false;
+            return category;
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(
+                "An exception has occurred while trying to retrieve section progress with the following message - {message}",
+                e.Message);
+            category.RetrievalError = true;
+            return category;
+        }
+    }
+
+    private async Task<CategorySectionRecommendationDto> GetSectionRecommendation(ISectionComponent section, SectionStatusDto? sectionStatus)
+    {
+        var sectionMaturity = sectionStatus?.Maturity;
+
+        if (string.IsNullOrEmpty(sectionMaturity)) return new CategorySectionRecommendationDto();
+
+        var recommendation =
+            await _getSubTopicRecommendationQuery.GetRecommendationsViewDto(section.Sys.Id, sectionMaturity);
+
+        if (recommendation == null)
+        {
+            _logger.LogError("No Recommendation Found: Section - {sectionName}, Maturity - {sectionMaturity}", section.Name, sectionMaturity);
+
+            return new CategorySectionRecommendationDto
+            {
+                NoRecommendationFoundErrorMessage = $"Unable to retrieve {section.Name} recommendation"
+            };
+        }
+
+        if (section.InterstitialPage?.Slug == null)
```

Comment:
Is this the most appropriate place for this?

Reply:
In fact - this appears to be duplicated by LogErrorWithUserFeedback, have updated that to include section name and deleted this